torch/nn/modules/module.py

```diff
@@ -281,22 +281,12 @@ def register_module_full_backward_pre_hook(
         This adds global state to the `nn.module` module
         and it is only intended for debugging/profiling purposes.
 
-    The hook will be called every time the gradients for the module are computed.
-    The hook should have the following signature::
+    Hooks registered using this function function in the same way as those
+    registered by :meth:`torch.nn.Module.register_backward_pre_hook`.
```

Comment:
The full hooks should be the ones that we want everyone to use. We shouldn't document them based on these other ops we don't want users to use.

If you want to dedup, you can update the non-full hooks to say not to use them, point to the full version for full doc and mention the difference.

Reply:
Whoops typo

Reply:
> If you want to dedup, you can update the non-full hooks to say not to use them, point to the full version for full doc and mention the difference.

Yeah for non-full global hooks I think we already do that today.